### ISSUE TITLE
NSFW Allowed?

### ISSUE THREAD

**Issue by AnnoSpmz** [carefree0910/carefree-creator#22]
is it possible to disable the nsfw checker? with an argument like --no_nsfw_checker, or any other way? (for scientific reasons)

**Comment by carefree0910** [comment]
Hi! If you can deploy locally, there are two ways to do it:
1. Refer to [this issue](https://github.com/carefree0910/carefree-creator/issues/19).
2. You can actually call the APIs directly, which will give you back the image `Response` without any nsfw checker!

> The API documentation locates at something like `{host}/redoc`, where the `host` is your domain url. 😉

**Comment by AnnoSpmz** [comment]
> Hi! If you can deploy locally, there are two ways to do it:
> 
> 1. Refer to [this issue](https://github.com/carefree0910/carefree-creator/issues/19).
> 2. You can actually call the APIs directly, which will give you back the image `Response` without any nsfw checker!
> 
> > The API documentation locates at something like `{host}/redoc`, where the `host` is your domain url. 😉

Hello! Thanks, but I can't find any images logged under the network tab, it only seems like api requests to `gw.nolibox.com`. Am I doing something wrong? I will look into the second option you mentioned, although I don't really know how to do that. Will read the documentation at `/redoc` I guess, thanks :)

**Comment by carefree0910** [comment]
> > Hi! If you can deploy locally, there are two ways to do it:
> > 
> > 1. Refer to [this issue](https://github.com/carefree0910/carefree-creator/issues/19).
> > 2. You can actually call the APIs directly, which will give you back the image `Response` without any nsfw checker!
> > 
> > > The API documentation locates at something like `{host}/redoc`, where the `host` is your domain url. 😉
> 
> Hello! Thanks, but I can't find any images logged under the network tab, it only seems like api requests to `gw.nolibox.com`. Am I doing something wrong? I will look into the second option you mentioned, although I don't really know how to do that. Will read the documentation at `/redoc` I guess, thanks :)

Hi! Looks like you have not deployed the APIs locally before you look into the network tab, have you installed this package, or used the Google Colab to deploy one?

**Comment by AnnoSpmz** [comment]
> > > Hi! If you can deploy locally, there are two ways to do it:
> > > 
> > > 1. Refer to [this issue](https://github.com/carefree0910/carefree-creator/issues/19).
> > > 2. You can actually call the APIs directly, which will give you back the image `Response` without any nsfw checker!
> > > 
> > > > The API documentation locates at something like `{host}/redoc`, where the `host` is your domain url. 😉
> > 
> > 
> > Hello! Thanks, but I can't find any images logged under the network tab, it only seems like api requests to `gw.nolibox.com`. Am I doing something wrong? I will look into the second option you mentioned, although I don't really know how to do that. Will read the documentation at `/redoc` I guess, thanks :)
> 
> Hi! Looks like you have not deployed the APIs locally before you look into the network tab, have you installed this package, or used the Google Colab to deploy one?

Oh right! I'm stupid, I was using your server, without deploying it myself. Sorry, also, I'm really new so you will have to excuse me, but what is Google Colab and how does it work? Can I run code in google colab? Is that in the cloud or locally? Do I have to download anything there? Thank you very much

**Comment by carefree0910** [comment]
Hi! You can treat Google Colab as a 'online code runner with GPUs', which means:
1. yes, you can run codes in it.
2. in the cloud.
3. yes you have to download my models there, but the models will be downloaded to their (Google's) cloud server, and will not occupy your disk space.

And here's the Colab link: https://colab.research.google.com/github/carefree0910/carefree-creator/blob/dev/tests/server.ipynb, you can try it out to see if everything goes right!

**Comment by AnnoSpmz** [comment]
Okay! Thank you very much :)

**Comment by AnnoSpmz** [comment]
Hey! I started up the Google Colab version, but it seems like what you said in that other issue, about it being visible in the networks tab, didn't work... It works for the regular images, but not for the flagged ones. Although the logged png I get in the network tab on the normal ones already seems uploaded on your cdn, so I get that the flagged ones aren't there. The other stuff just seems like API requests to my webserver and to your cdn. Are you sure that we're able to see the raw generated on the networks tab? Thanks!

**Comment by carefree0910** [comment]
Yes, I'm pretty sure, because the API service can only return raw images, which will be uploaded to the cloud at the WebUI. So before uploading, the raw images have to be sent to the WebUI, so you should be able to see the images!

> - The requests correspond to these raw images in the network tab will be displayed as `sd`.
> - You may fail to see the raw image from the network tab directly, but you can copy the request into `curl` or other format you like and utilize your cmd/postman to fetch the outputs!
